# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [65]:
import pandas as pd
import numpy as np
from scipy import stats
import seaborn as sns 
import matplotlib
import matplotlib.pyplot as plt
import pylab as pl
%matplotlib inline



In [66]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [67]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [68]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [69]:
# number of callbacks for white-sounding names
sum(data[data.race=='w'].call)

235.0

# 1. What test is appropriate for this problem? Does CLT apply?

In [95]:
print(len(data[data.race=='b']))
print(len(data[data.race=='w']))

2435
2435


In general, the sample size, n, should be above about 30 in order for the Central Limit Theorem to be applicable.Since we have pretty large sample size, good approximations can be made to get closer to a normal distribution by applying CLT. 

# 2. What are the null and alternate hypotheses?

In [87]:
# Calculate call backs for white and black sounding names in terms of mean
w_data = data[data.race=='w']
b_data = data[data.race=='b']
Mw= w_data['call'].mean()
Mb= b_data['call'].mean()
print('Sample mean of call backs for white sounding %.3f' %Mw)
print('Sample mean of call backs for black sounding %.3f' %Mb)

Sample mean of call backs for white sounding 0.097
Sample mean of call backs for black sounding 0.064


In [86]:
# Difference between calculated means for each type (w and b) of call backs
print('Mw - Mb= %.3f' %(Mw-Mb))

Mw - Mb= 0.032


- Null hypotheses/Ho: Mw - Mb=0 - There is no difference between callbacks for white soundings and blacks soundings. 
- Alternate hypotheses/H1 :Mw - Mb≠0 - There is a difference between callbacks for white soundings and black soundings.

# 3. Compute margin of error, confidence interval, and p-value.

In [104]:
#Apply two sample t-test
w_callback = data[data.race=='w'].call
b_callback = data[data.race=='b'].call
two_sample=stats.ttest_ind(w_callback, b_callback)
print('The results of the two sample t-test are: \n\tt-value = {:4.3f}\n\tp-value = {:4.5f}'.format(two_sample[0],two_sample[1]))

The results of the two sample t-test are: 
	t-value = 4.115
	p-value = 0.00004


Based on the pvalue < 0.05, we reject the null hyphothesis and there is a difference between callbacks for two group of means.

In [149]:
Nw=2435
Nb=2435
df=Nw + Nb -2
z_critical = stats.norm.ppf(q = 0.975)  # Get the z-critical value*

print("z-critical value:")              # Check the z-critical value
print(z_critical)                        

z-critical value:
1.95996398454


In [153]:
from math import sqrt
Mw= w_data['call'].mean()
Mb= b_data['call'].mean()
diff_mean= Mw-Mb
#Margin of error, MoE= z_critical * standard error
SE = sqrt ( stdw**2 / Nw + stdb**2 / Nb )
MoE= 1.96 * SE
print('The margin of error %.3f' %MoE)
#confidence_interval = (diff_mean - margin_of_error, diff_mean + margin_of_error) 
print ('\nThe difference between groups is {:3.4f} [{:3.3f} to {:3.3f}] (mean [95% CI])'.format(diff_mean, diff_mean - MoE, diff_mean + MoE))

The margin of error 0.015

The difference between groups is 0.0320 [0.017 to 0.047] (mean [95% CI])


# 4. Write a story describing the statistical significance in the context or the original problem.



Based on the given p value and 95% confidance interval, there is a statistical significance between two groups of calling backs. The white sounding names are getting 3.2 % more calls than black sounding names between 0.047 and 0.017 calls.

# 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

In this analysis, we just examined whether race has a significant impact on the rate of callbacks for resumes. On the other hand, after creating a dummy variable as a dependent variable, influences of all other or some factors on race/id can be examined. Thus, it is early to say that the race/name is the most important factor in callback success.  